validator/client/runner.go

```diff
@@ -67,9 +71,17 @@ func run(ctx context.Context, v Validator) {
 	if err := v.WaitForSync(ctx); err != nil {
 		log.Fatalf("Could not determine if beacon node synced: %v", err)
 	}
-	if err := v.WaitForActivation(ctx); err != nil {
+
+	val, ok := v.(*validator)
+	if !ok {
+		log.Fatalf("Could not wait for validator activation: could not instantiate validator")
+	}
+	accountsChangedChan := make(chan struct{})
+	go val.handleAccountsChanged(ctx, accountsChangedChan)
```

Comment:
Don't feel too great about this, because we wanted to keep this `run` function abstract, using the interface only. Doing this type assertion goes against the design of this function

Reply:
Yeah, this doesn't look too good. I will add a new function to the interface, so that no type assertions are required.